src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -926,3 +928,198 @@ def acquireTiledArea(streams, stage, area, overlap=0.2, settings_obs=None, log_p
+                               log_path, zlevels,
+                               registrar, weaver, focusing_method)
+    future.task_canceller = task.cancel  # let the future cancel the task
+
+    future.set_progress(end=task.estimate_time() + time.time())
+    # connect the future to the task and run in a thread
+    executeAsyncTask(future, task.run)
+
+    return future
+
+
+class AcquireOverviewTask(object):
+    """
+    Create a task to run autofocus and tiled acquisition for each area in the list of areas
+    """
+
+    def __init__(self, streams, stage, areas, focus, ccd, future, overlap=0.2, settings_obs=None, log_path=None,
+                 zlevels=None,
+                 registrar=REGISTER_GLOBAL_SHIFT, weaver=WEAVER_MEAN, focusing_method=FocusingMethod.NONE):
+        # site and feature means the same
+        self._stage = stage
+        self._future = future
+        if future is not None:
+            self._future.running_subf = model.InstantaneousFuture()
+            self._future._task_lock = threading.Lock()
+        self.streams = streams
+        self.areas = areas  # list of areas
+        self._ccd = ccd  # TODO to delete after ccd.data components is used from streams in do_autofocus_roi
+        self._focus = focus
+        self.focus_rng = self._focus.getMetadata().get(model.MD_POS_ACTIVE_RANGE, None)
+
+        if self.focus_rng is None:
+            # No absolute range is available, so use a relative range (e.g. as on METEOR
+            self.focus_rng = self._focus.getMetadata().get(model.MD_SAFE_REL_RANGE, SAFE_REL_RANGE_DEFAULT)
+            self.focus_rng = (self.focus_rng[0] + self._focus.position.value["z"],
+                              self.focus_rng[1] + self._focus.position.value["z"])
+
+        self.conf_level = 0.8
+        self.focusing_method = focusing_method
+        # The number of focus points are constant for now
+        # TODO: make it variable according to the size of the ROI, is it useful?
+        self.n_focus_points = (3, 3)
+        self._overlap = overlap
+        self._settings_obs = settings_obs
+        self._log_path = log_path
+        self._zlevels = zlevels
+        self._registrar = registrar
+        self._weaver = weaver
+
+    def cancel(self, future):
+        """
+        Canceler of acquisition task.
+        """
+        logging.debug("canceling acquisition overview...")
+
+        with future._task_lock:
+            if future._task_state == FINISHED:
+                return False
+            future._task_state = CANCELLED
+            future.running_subf.cancel()
+            logging.debug("acquisition overview cancelled.")
+        return True
+
+    def estimate_time(self, roi_idx=0, actual_time_per_roi=None) -> float:
+        """
+        Estimates the time for the rest of the acquisition.
+
+        :param roi_idx: (int) index of the current roi
+        :param actual_time_per_roi: (float) the actual time spent for the current roi
```

Comment:
this is not the actual time spent for the current roi, but the average time spent for all rois acquired until now.